Inc/HALALMock/Models/PinModel/Pin.hpp

```diff
@@ -6,156 +6,158 @@
 
-enum GPIOPin : uint16_t{
-	PIN_0 = 0x0001,
-	PIN_1 = 0x0002,
-	PIN_2 = 0x0004,
-	PIN_3 = 0x0008,
-	PIN_4 = 0x0010,
-	PIN_5 = 0x0020,
-	PIN_6 = 0x0040,
-	PIN_7 = 0x0080,
-	PIN_8 = 0x0100,
-	PIN_9 = 0x0200,
-	PIN_10 = 0x0400,
-	PIN_11 = 0x0800,
-	PIN_12 = 0x1000,
-	PIN_13 = 0x2000,
-	PIN_14 = 0x4000,
-	PIN_15 = 0x8000,
-	PIN_ALL = 0xFFFF
+enum GPIOPin : uint16_t {
+    PIN_0 = 0x0001,
+    PIN_1 = 0x0002,
+    PIN_2 = 0x0004,
+    PIN_3 = 0x0008,
+    PIN_4 = 0x0010,
+    PIN_5 = 0x0020,
+    PIN_6 = 0x0040,
+    PIN_7 = 0x0080,
+    PIN_8 = 0x0100,
+    PIN_9 = 0x0200,
+    PIN_10 = 0x0400,
+    PIN_11 = 0x0800,
+    PIN_12 = 0x1000,
+    PIN_13 = 0x2000,
+    PIN_14 = 0x4000,
+    PIN_15 = 0x8000,
+    PIN_ALL = 0xFFFF
 };
 
 enum GPIOPort : uint32_t {
-	PORT_A = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0000UL,
-	PORT_B = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0400UL,
-	PORT_C = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0800UL,
-	PORT_D = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0C00UL,
-	PORT_E = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1000UL,
-	PORT_F = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1400UL,
-	PORT_G = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1800UL,
-	PORT_H = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1C00UL,
+    PORT_A = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0000UL,
+    PORT_B = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0400UL,
+    PORT_C = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0800UL,
+    PORT_D = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x0C00UL,
+    PORT_E = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1000UL,
+    PORT_F = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1400UL,
+    PORT_G = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1800UL,
+    PORT_H = DOMAIN3_ADVANCED_HIGH_PERFORMANCE_BUS1 + 0x1C00UL,
 };
 
-enum AlternativeFunction: uint8_t {
-	AF0 = 0x00,
-	AF1 = 0x01,
-	AF2 = 0x02,
-	AF3 = 0x03,
-	AF4 = 0x04,
-	AF5 = 0x05,
-	AF6 = 0x06,
-	AF7 = 0x07,
-	AF8 = 0x08,
-	AF9 = 0x09,
-	AF10 = 0x0A,
-	AF11 = 0x0B,
-	AF12 = 0x0C,
-	AF13 = 0x0D,
-	AF14 = 0x0E,
-	AF15 = 0x0F,
+enum AlternativeFunction : uint8_t {
+    AF0 = 0x00,
+    AF1 = 0x01,
+    AF2 = 0x02,
+    AF3 = 0x03,
+    AF4 = 0x04,
+    AF5 = 0x05,
+    AF6 = 0x06,
+    AF7 = 0x07,
+    AF8 = 0x08,
+    AF9 = 0x09,
+    AF10 = 0x0A,
+    AF11 = 0x0B,
+    AF12 = 0x0C,
+    AF13 = 0x0D,
+    AF14 = 0x0E,
+    AF15 = 0x0F,
 };
 
-enum OperationMode{
-	NOT_USED,
-	INPUT,
-	OUTPUT,
-	ANALOG,
-	EXTERNAL_INTERRUPT_RISING,
-	EXTERNAL_INTERRUPT_FALLING,
-	TIMER_ALTERNATE_FUNCTION,
-	ALTERNATIVE,
+enum OperationMode {
+    NOT_USED,
+    INPUT,
+    OUTPUT,
+    ANALOG,
+    EXTERNAL_INTERRUPT_RISING,
+    EXTERNAL_INTERRUPT_FALLING,
+    TIMER_ALTERNATE_FUNCTION,
+    ALTERNATIVE,
 };
 
-enum PinState{
-	OFF,
-	ON
-};
+enum PinState { OFF, ON };
 
 enum class PinType {
-	NOT_USED,
-	DigitalOutput,
-	DigitalInput,
-	PWM,
-	DualPWM,
-	ADC
-	// TODO: Add more types
+    NOT_USED,
+    DigitalOutput,
+    DigitalInput,
+    PWM,
```

Comment:
remove this changes form this branch, this PR should just include what it adds, from my perspective if you have dependency on this merge this PR into the branch that adds this to develop and when that branch gets merged rebase the PR to develop

Reply:
But to add SPI mock I had to create a new type in emulated pin. It is directly related to this PR

Reply:
it is just a linter issue :))

Reply:
But the whole file is changed

Reply:
Someone made a merge in development branch messing up everything and now seems that this file is totally changed, but is not my fault!